File: 1146-greatest-common-divisor-of-strings/1146-greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        def ucln(a,b):
            if b==0:
                return a
            return ucln(b,a%b)
        len_kq = ucln(len(str1),len(str2))
        candidate = str1[:len_kq]
        if str1 == candidate * (len(str1)//len(candidate)) and str2 == candidate * (len(str2)//len(candidate)):
            return candidate
        return ""

File: 1146-greatest-common-divisor-of-strings/1146-greatest-common-divisor-of-strings.cpp
int ucln(int a,int b) {
            if (b==0)
                return a;
            return ucln(b,a%b);
        }
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        int len_kq = ucln(str1.length(),str2.length());
        string candidate = str1.substr(0,len_kq);
        string temp1 = "";
        for(int i =0;i<str1.length()/len_kq;i++) {
            temp1 += candidate;
        }
        string temp2 = "";
        for(int i =0;i< str2.length()/len_kq;i++) {
            temp2 += candidate;
        }
        if ((str1 == temp1) && (str2 == temp2))
            return candidate;
        else {
            return "";
        }
    }
};